code/modules/mob/living/simple_animal/revenant/revenant_abilities.dm

```diff
@@ -154,6 +164,81 @@
 			for(var/obj/machinery/mach in T.contents)
 				if(prob(10))
 					mach.emag_act(null)
-	empulse(user.loc, 3, 5)
+				else
+					mach.emp_act(1)
 	user.reveal(reveal)
-	user.stun(stun)
\ No newline at end of file
+	user.stun(stun)
+
+
+//Fling Object: Finds the item nearby with the highest throwforce, then multiplies it and throws it at the target.
+/obj/effect/proc_holder/spell/targeted/revenant_throw_item
+	name = "Fling Object (5E)"
+	desc = "Finds the nearest item that hits the hardest when thrown, and throws it really hard."
+	panel = "Revenant Abilities" //Not locked by default
+	charge_max = 50
+	clothes_req = 0
+	action_icon_state = "summons"
+	action_background_icon_state = "bg_revenant"
+	var/stun = 30
```

Comment:
Doesn't actually stun because there's no `user.stun(stun)` in the cast, also making it stun is kind of weird if it doesn't reveal